### ISSUE TITLE
file out node doesn't create file in multi-level sub-directories

### ISSUE THREAD

**Issue by andysc** [node-red/node-red#1489]
To help us understand the issue, please fill-in as much of the following information as you can:

### What are the steps to reproduce?

```
[{"id":"e29214f8.aa55c","type":"inject","z":"33edeb5d.10333c","name":"","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"x":200,"y":160,"wires":[["a054f716.5a85b"]]},{"id":"a054f716.5a85b","type":"file","z":"33edeb5d.10333c","name":"","filename":"fred/jim/file.txt","appendNewline":true,"createDir":true,"overwriteFile":"true","x":400,"y":160,"wires":[]}]
```
### What happens?

`"failed to write to file: Error: ENOENT: no such file or directory, open 'fred/jim/file.txt'"`
Directories fred and jim are created, but file.txt is not created in fred/jim/

### What do you expect to happen?

I expect all the necessary sub-directories to be created, and then the file to be created in the lowest level directory.

### Please tell us about your environment:

- [ ] Node-RED version: v0.17.5
- [ ] node.js version: v6.11.5
- [ ] npm version: 3.10.10
- [ ] Platform/OS: Mac and Ubuntu
- [ ] Browser: Safari

**Comment by rcruicks** [comment]
looks like 50-file uses fs-extra - package.json has fs-extra:1.0.0 -- current version in 4.0.3, which includes fixes to ensureDir/mkdirs
updated package.json: fsextra:4.0.3 and grunt tests passed

**Comment by knolleary** [comment]
Fixed in 0.18 branch.

Needed a bit more than the update to fs-extra. The work to create the parent directories was being done asynchronously to the creation of the file - so was entirely possible the parent directories hadn't finished being created when it came to write the file.

**Comment by andysc** [comment]
Thanks Nick :)
